realm/realm-library/src/androidTest/java/io/realm/util/ExceptionHolder.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.util;
+
+import junit.framework.AssertionFailedError;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+import static junit.framework.Assert.fail;
+
+/**
+ * Utility class for wrapping exceptions from background threads in unit tests.
+ * It makes patterns like this possible:
+ *
+ * {@code
+ *
+ *   final ExceptionHolder bgError = new ExceptionHolder();
+ *   new Thread(new Runnable() {
+ *      @Override
+ *      public void run() {
+ *          try {
+ *              // Error prone code ...
+ *          } catch (Exception e) {
+ *              bgError.setException(e);
+ *          }
+ *      }
+ *   }).start()
```

Comment:
Yes, you need som kind of latch to detect when the thread is done. This code sample could be more explicit about that, true.